kolena/workflow/test_case.py

```diff
@@ -332,3 +354,91 @@ def edit(self, reset: bool = False) -> Iterator[Editor]:
         test_case_data = from_dict(data_class=CoreAPI.EntityData, data=complete_res.json())
         self._populate_from_other(self._create_from_data(test_case_data))
         log.success(f"edited test case '{self.name}' (v{self.version})")
+
+    @classmethod
+    def init_many(
+        cls,
+        data: List[Tuple[str, List[Tuple[TestSample, GroundTruth]]]],
+        reset: bool = False,
+    ) -> List["TestCase"]:
+        """
+        !!! note "Experimental"
+
+            This function is considered **experimental**, so beware that it is subject to changes
+            even in patch releases.
+
+        Create, load or edit multiple test cases.
+
+        ```python
+        test_cases = TestCase.init_many([
+            ("test-case 1", [(test_sample_1, ground_truth_1), ...]),
+            ("test-case 2", [(test_sample_2, ground_truth_2), ...])
+        ])
+
+        test_suite = TestSuite("my test suite", test_cases=test_cases)
+        ```
+
+        Changes are committed to the Kolena platform together. If there is an error, none of the edits would take
+        effect.
+
+        :param data: A list of tuples where each tuple is a test case name and a set of test samples and ground truths
+                     tuples for the test case.
+        :param reset: If a test case of the same name already exists, overwrite with the provided test_samples.
+        :return: The test cases.
+
+
+        """
+
+        if not hasattr(cls, "workflow"):
+            raise NotImplementedError(f"{cls.__name__} must implement class attribute 'workflow'")
```

Comment:
Is this necessary?

Reply:
I shot myself in the foot when writing test like this `TestCase.init_many()`, when in fact I should use `DummyTestCase.init_many`. The method needs class workflow information to initialize test-case instances.

Reply:
Isn't this validated via `_validate_test_samples`?

Reply:
Since `_validate_test_samples` access `cls.workflow` directly instead of checking its existence, relying on it would see the same exception as `TestCase._init_many()`.